sitemap/sitemap_test.go

```diff
@@ -0,0 +1,110 @@
+package sitemap
+
+import (
+	"bytes"
+	"embed"
+	"encoding/json"
+	"jobs"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"utils"
+
+	"github.com/gin-gonic/gin"
+	"github.com/jmoiron/sqlx"
+	"github.com/tj/assert"
+)
+
+//go:embed templates/career-email-template.html
+var templateFS embed.FS
+
+var repo *SitemapRepository
+var careerRepo *jobs.CareerRepository
+var err error
+var testDB *sqlx.DB
+var testRequests []utils.TestRequest
+
+const (
+	GET_SITEMAP_API_URL = "/sitemap?baseUrl=http://localhost:8080"
+)
+
+func Test_init(t *testing.T) {
+	repo, err = initializeRepo()
+	assert.Nil(t, err)
+	testRequests = []utils.TestRequest{
+		{
+			Url:               GET_SITEMAP_API_URL,
+			Method:            "GET",
+			Headers:           nil,
+			Body:              nil,
+			ResponseCode:      http.StatusOK,
+			ResponseTypeArray: false,
+			ExpectedData:      []interface{}{expectedSitemapData()},
+		},
+	}
+}
+
+func TestAllAPIs(t *testing.T) {
+
+	asserts := assert.New(t)
+	engine := gin.New()
+
+	setUpRouter(engine)
+
+	for _, testData := range testRequests {
+
+		w := httptest.NewRecorder()
+		var req *http.Request
+
+		if testData.Body != nil {
+			requestByte, _ := json.Marshal(testData.Body)
+			reqBodyData := bytes.NewReader(requestByte)
+			req, err = http.NewRequest(testData.Method, testData.Url, reqBodyData)
+		} else {
+			req, err = http.NewRequest(testData.Method, testData.Url, nil)
+		}
+
+		asserts.NoError(err)
+
+		engine.ServeHTTP(w, req)
+		assert.Equal(t, testData.ResponseCode, w.Code)
+
```

Comment:
Can you once try with the following flow to assert XML bytes data? 

-  Convert request body to bytearray
- Then unmarshal it using xml